### PR TITLE
Activities

### PR DESCRIPTION
**Pull request by Victorma**
+ Changed sessions to activities
+ Classes are no longer needed
+ Toolbar access fixed
+ Routes updated
+ Improved visuals
+ And many other things...

### CODE REVIEW

File: app/public/js/controllers/activities.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+'use strict';
+
+angular.module('activitiesApp', ['ngStorage', 'services'])
+    .controller('ActivitiesCtrl', ['$scope', '$location', '$http', '$window', '$localStorage', 'Games', 'Versions',
+        'Activities', 'Role', 'CONSTANTS', 'QueryParams',
+        function ($scope, $location, $http, $window, $localStorage, Games, Versions, Activities, Role, CONSTANTS, QueryParams) {
+            $scope.$storage = $localStorage;
+            $scope.activity = {};
+            $scope.class = {};
+            $scope.game = {};
```
Comment (gorco) [inline app/public/js/controllers/activities.js]: $scope.game and $scope.class are not used in this controller

File: app/public/js/controllers/activity.js [hunk 2]
```diff
@@ -18,7 +18,7 @@
```
Comment (gorco) [inline app/public/js/controllers/activity.js]: If we use Activities controller now, the old Activity controller has to dissapear

File: app/views/layout.jade [hunk 3]
```diff
@@ -61,14 +61,14 @@ html
         script(src='js/controllers/devVisualizator.js')
         script(src='js/controllers/data.js')
         script(src='js/controllers/analytics.js')
-        script(src='js/controllers/session.js')
+        script(src='js/controllers/activity.js')
```
Comment (gorco) [inline app/views/layout.jade]: we have to replace activity by activities (remove the activity reference)

File: app/views/toolbar.jade [hunk 4]
```diff
@@ -41,15 +41,24 @@ nav.navbar.navbar-inverse.navbar-top
                         ul.dropdown-menu(aria-labelledby='dropdownGames')
                             li(ng-repeat='game in games')
                                 a(ng-if='isDeveloper()' ng-href='data?game={{game._id}}' ng-click='setSelectedGame(game)') {{game.title}}
-                                a(ng-if='!isDeveloper()' ng-href='class?game={{game._id}}' ng-click='setSelectedGame(game)') {{game.title}}
+                                a(ng-if='!isDeveloper()' ng-href='gameactivity?game={{game._id}}' ng-click='setSelectedGame(game)') {{game.title}}
                 li.dropdown
-                    li(ng-if='isUser()&&isTeacher()&&hasSessions()')
+                    li(ng-if='isUser()&&hasClasses()')
```
Comment (gorco) [inline app/views/toolbar.jade]: we should check the role (we don't want that developers to see classes)

File: app/views/toolbar.jade [hunk 5]
```diff
@@ -41,15 +41,24 @@ nav.navbar.navbar-inverse.navbar-top
...
                         a#dropdownClasses.btn(data-toggle='dropdown' aria-haspopup='true' aria-expanded='false')
                             | Classes
                             span.caret
                         ul.dropdown-menu(aria-labelledby='dropdownClasses')
-                            li(ng-repeat='session in sessions')
-                                a(ng-href="#" ng-click='setSelectedSession(session, "data")') {{session.name}}
+                            li(ng-repeat='class in classes')
+                                a(ng-if='isDeveloper()' ng-href="#" ng-click='setSelectedClass(class, "data")') {{class.name}}
```
Comment (gorco) [inline app/views/toolbar.jade]: The developers can't see classes

File: app/views/data/realtime/realtime.jade [hunk 6]
```diff
@@ -16,7 +16,7 @@
     limitations under the License.
  
 link(rel="stylesheet", href="css/realtime.css")
-.container-fluid.height100(ng-controller="SessionCtrl" role="tabpanel" style="position:relative")
+.container-fluid.height100(ng-controller="ActivityCtrl" role="tabpanel" style="position:relative")
```
Comment (gorco) [inline app/views/data/realtime/realtime.jade]: ActivitiesCtrl

**Review by gorco — CHANGES_REQUESTED** [review]
Great work, some commets pending.
Awesome changes at home view

File: app/views/toolbar.jade (hunk 4 above)
Comment (Victorma) [reply app/views/toolbar.jade]: Fixed in: https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/5353294c7dba6c4720c637a730bf89ca7c7cbfb0

File: app/views/toolbar.jade (hunk 5 above)
Comment (Victorma) [reply app/views/toolbar.jade]: Fixed in: https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/5353294c7dba6c4720c637a730bf89ca7c7cbfb0

File: app/views/data/realtime/realtime.jade (hunk 6 above)
Comment (Victorma) [reply app/views/data/realtime/realtime.jade]: Fixed in: https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/0edb046efc1121fa233acb2cd3d9afd7dd93ff4f

File: app/views/layout.jade (hunk 3 above)
Comment (Victorma) [reply app/views/layout.jade]: Fixed in: https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/572bff82cfdb94cdb0df0d32cef2344ce44ca856 and https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/232f3d0b7236dd1f7e4f23a330d256a0370a11aa

File: app/public/js/controllers/activity.js (hunk 2 above)
Comment (Victorma) [reply app/public/js/controllers/activity.js]: The activity-list controller manages the creation and deletion. The activity manages the modification and realtime start and stop.

File: app/public/js/controllers/activities.js (hunk 1 above)
Comment (Victorma) [reply app/public/js/controllers/activities.js]: Removed in: https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/572bff82cfdb94cdb0df0d32cef2344ce44ca856

**Comment by gorco** [comment]
I'm testing the front, but when I create a new game (also if I open a game) the system throws these errors...
Backend:
backerror
Frontend
fronterror

**Comment by gorco** [comment]
The developer visualizations don't load
frontdev

**Comment by gorco** [comment]
As teacher, Into a Game, if I try create a activity, I can't because the frontend says that I don't have selected a game.
fronteaerror

**Pull request closed without merging**